### PR TITLE
Add error logs on parsing and validation errors.

### LINKED ISSUE #364: Capture jetty standard out/error
Either log it (preferably) or print it.  We must have lost this feature when we stopped using lein-ring.

One way to do this would be to add a custom middleware that catches exceptions and logs them before re-throwing them.

### PR DESCRIPTION
**Pull request by yogsototh**
See: https://github.com/metosin/compojure-api/wiki/Exception-handling
	and: https://github.com/metosin/compojure-api/blob/master/src/compojure/api/exception.clj

Should close #364

### CODE REVIEW

**Comment by polygloton** [comment]
I think you should override the default as well, so that we can be explicit about it.

**Comment by polygloton** [comment]
Also check for ExceptionInfo and included the ex-data in the log message if there is any.

File: src/ctia/http/exceptions.clj [hunk 1]
```diff
@@ -1,28 +1,40 @@
...
+  (logging/log! :error e (ex-message e))
   (ex/response-validation-handler e data request))
 
-
+(defn default-error-handler
+  "Handle default error"
+  [^Exception e data request]
+  (logging/log! :error e (ex-message e))
+  (internal-server-error {:type "unknown-exception"
+                          :class (.getName (.getClass e))}))
```
Comment (quoll) [inline src/ctia/http/exceptions.clj]: It's usually better to use clojure.core functions when possible. So:
`   ... :class (.getName (class e)) ...`

Comment (yogsototh) [inline src/ctia/http/exceptions.clj]: I've copy/pasted from the metosin code as the goal was to keep mostly the same function but only change its logs. But yes, I agree, `class` is better.

**Pull request merged**